packages/evm/src/hybridStack.ts

```diff
@@ -0,0 +1,314 @@
+import { bigIntToBytes, bytesToBigInt } from '@ethereumjs/util'
+
+import { ERROR, EvmError } from './exceptions.js'
+
+import type { EVMStack } from './types.js'
+
+/**
+ * Hybrid stack implementation for the EVM, containing either bytes (preferred) or bigint values.
+ *
+ * This stack implementation is significantly more performant than the single type stack. Atm it is however
+ * not activated by default for backwards compatibility reasons.
+ *
+ * In the current EVM version this stack performs significantly worse than the hybrid type stack on the EVM
+ * `step` event (if used), since there is mapping to a single type stack needed on each step.
```

Comment:
This is fine. We might want to note that if one listens to `step` then EVM performance will be much slower due to the objects it has to generate. `step` event should be somewhat seen as a `debug` mode event.